### ISSUE TITLE
feat(output): add JUnit XML output format

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#271]
## Overview

Add a `junit` output format so gomarklint results can be consumed by CI/CD test report dashboards. JUnit XML is the de facto standard for structured test results across GitHub Actions, Jenkins, CircleCI, and GitLab CI.

## Problem

gomarklint currently supports `text` and `json` output. Neither integrates with the test report UIs built into CI platforms. With JUnit output, broken links and lint violations surface in the Checks tab of a PR — no log-diving required.

## How it would look

```
gomarklint --output junit . > results.xml
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="gomarklint" tests="3" failures="1" time="0.42">
  <testsuite name="docs/guide.md" tests="2" failures="1">
    <testcase name="line 12: Link unreachable: https://example.com/broken" classname="docs/guide.md">
      <failure message="Link unreachable: https://example.com/broken" type="error"/>
    </testcase>
    <testcase name="line 5: heading-increment" classname="docs/guide.md"/>
  </testsuite>
  <testsuite name="README.md" tests="1" failures="0">
    <testcase name="README.md" classname="README.md"/>
  </testsuite>
</testsuites>
```

GitHub Actions usage:

```yaml
- run: gomarklint --output junit . > results.xml
- uses: actions/upload-artifact@v4
  if: always()
  with:
    name: gomarklint-results
    path: results.xml
```

## Implementation

The output package already defines a clean `Formatter` interface:

```go
type Formatter interface {
    Format(w io.Writer, result *Result) error
}
```

`app.go` selects the formatter based on `cfg.OutputFormat` (currently `"text"` or `"json"`).

### Files to create

- `internal/output/junit.go` — `JUnitFormatter` implementing `Formatter`

### Files to modify

- `internal/app/app.go` — add `"junit"` branch in `formatOutput`
- `internal/config/config.go` — add `"junit"` to valid `output_format` values
- docs and CLI help text

### JUnit mapping

| gomarklint concept | JUnit element |
|---|---|
| Run | `<testsuites>` |
| File | `<testsuite>` |
| No violations in file | `<testcase>` (passing) |
| Violation | `<testcase>` + `<failure>` |
| `elapsed_ms` | `time` attribute (seconds) |

Files with no violations emit a single passing `<testcase>` so the suite shows up in the report rather than being absent.

## Acceptance criteria

- [ ] `--output junit` (or `"output_format": "junit"` in config) produces valid JUnit XML
- [ ] Each file maps to a `<testsuite>`; each violation maps to a `<testcase>` with `<failure>`
- [ ] Files with no violations emit a single passing `<testcase>`
- [ ] `time` attribute reflects actual elapsed time in seconds
- [ ] Output passes standard JUnit XML schema validation
- [ ] Unit tests for `JUnitFormatter` covering no-error and error cases
- [ ] Invalid `output_format` value produces a clear error message

**Comment by shinagawa-web** [comment]
## Decision: closing this issue

After prototyping the JUnit XML formatter and integrating it into the CI pipeline, we evaluated whether it actually delivers the stated goal of surfacing violations in the Checks tab without log-diving.

### What we found

**JUnit XML is not natively supported by GitHub Actions.**
To display JUnit XML results in the Checks tab, a third-party action (e.g. `mikepenz/action-junit-report`) is required. Even with that action wired up, the check run body is empty for passing runs, and violation details do not surface as meaningful inline content — you still need to navigate into the job logs to see specifics.

**GitHub Annotations (`::error file=X,line=Y::message`) were also considered.**
These appear inline in the "Files changed" tab, but only for files that are part of the PR diff. Since gomarklint typically lints docs and README files that are not modified in a given PR, annotations on those files would not appear in the diff view. The benefit over the existing PR comment is minimal.

**The existing `comment-on-pr` feature already covers the use case.**
When `comment-on-pr: true` is set in the GitHub Action, violations are posted as a PR comment — immediately visible when the PR is opened, no navigation required. For link-check results specifically, knowing *which URL is broken* matters more than seeing it inline at the source line.

### Conclusion

The premise of this issue — that JUnit XML would enable a no-log-diving experience on GitHub Actions — does not hold. JUnit XML's value is for CI platforms with native test dashboard support (Jenkins, CircleCI, GitLab CI), which are not the primary target given that a GitHub Action is already provided.

Closing in favor of the existing PR comment feature.
